fossunited/foss_profiles/doctype/foss_user_profile/foss_user_profile.py

```diff
@@ -17,6 +17,13 @@ def set_route(self):
         self.route = self.username.lower().replace("-", "_")
 
     def get_context(self, context):
+        if self.is_anonymous and frappe.session.user not in [
+            "Administrator",
+            self.user,
+        ]:
+            frappe.redirect("/404")
+            return
```

Comment:
Not the right way, you have to use frappe.throw with Not found, check frappe code base for the same.